File: bank(csa2)/bankClient/src/Main.java
import org.apache.xmlrpc.XmlRpcException;

import java.net.MalformedURLException;
import java.util.Scanner;

public class Main {

    public static void main(String[] args) {

        try{

            BankClient client = new BankClient();

            BankCallback callback = new BankCallback(client);
            client.options(callback);

            Scanner scanner = new Scanner(System.in);
            String input = scanner.next();
            while (!input.equals("exit")){

                if (input.equals("login"))  client.login(scanner,callback);
                if (input.equals("register"))   client.register(scanner, callback);
                if (input.equals("accountBalance")) client.accountBalance(callback);
                if (input.equals("transfer"))   client.transfer(scanner, callback);
                if (input.equals("logout"))     client.logout(callback);
                if (input.equals("another")) client.anotherMethod(scanner);
                if (input.equals("options")) client.options(callback);
//                client.options(callback);
                input = scanner.next();
            }
        } catch (MalformedURLException e) {
            e.printStackTrace();
        } catch (XmlRpcException e) {
            e.printStackTrace();
        }
    }
}

File: bank(csa2)/bankClient/src/Main.kt
import com.jbosak.csa.bank.server.UserManager
import java.lang.reflect.Method

import java.util.Scanner

object Main {

    @JvmStatic
    fun main(args: Array<String>) {

        val client = BankClient()

        val scanner = Scanner(System.`in`)
        var input = scanner.next()
        while (input != "exit") {
            printParams(input)
            if (input == "login") client.login(scanner)
            if (input == "register") client.register(scanner)
            if (input == "accountBalance") client.accountBalance()
            if (input == "transfer") client.transfer(scanner)
            if (input == "logout") client.logout()
            if (input == "options") client.options()
            input = scanner.next()
        }
    }

    fun printParams(methodName: String){
        if(methodName == "transfer"){
            println("params: ")
            println( "String, int,")
        } else
        if(methodName != "accountBalance"){
            val method:List<Method?> = UserManager::class.java.methods
                    .filter { it.name == methodName}

            if (method.isNotEmpty() && method[0]?.parameters?.isNotEmpty()!!){
                println("params: ")

                method[0]!!.parameters
                        .map { it.type.simpleName }
                        .forEach { p -> print("$p, ") }
                println()
            }
        }

    }
}
